drivers/sensor/hs300x/hs300x.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2023 Ian Morris
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_hs300x
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "hs300x.h"
+
+#define HS300X_STATUS_MASK (BIT(0) | BIT(1))
+
+LOG_MODULE_REGISTER(HS300X, CONFIG_SENSOR_LOG_LEVEL);
+
+static int hs300x_start_measurement(const struct device *dev)
+{
+	const struct hs300x_config *cfg = dev->config;
+
+	/*
+	 * Initiate a measurement simply by sending 7-bit address followed
+	 * by an eighth bit set to 0 (write) and NO data.
+	 */
+	return i2c_write_dt(&cfg->bus, NULL, 0);
+}
+
+static int hs300x_read_sample(const struct device *dev, uint16_t *t_sample, uint16_t *rh_sample)
+{
+	const struct hs300x_config *cfg = dev->config;
+	uint8_t rx_buf[4];
+	int rc;
+
+	rc = i2c_read_dt(&cfg->bus, rx_buf, sizeof(rx_buf));
+	if (rc < 0) {
+		LOG_ERR("Failed to read data from device.");
+		return rc;
+	}
+
+	if ((rx_buf[3] & HS300X_STATUS_MASK) != 0) {
+		LOG_ERR("Stale data");
+		return -EIO;
+	}
+
+	*rh_sample = sys_get_be16(rx_buf);
+	*t_sample = sys_get_be16(&rx_buf[2]);
+
+	/* Remove status bits (only present in temperature value)*/
+	*t_sample >>= 2;
+
+	return 0;
+}
+
+static int hs300x_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct hs300x_data *data = dev->data;
+	int rc;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP &&
+	    chan != SENSOR_CHAN_HUMIDITY) {
+		return -ENOTSUP;
+	}
+
+	rc = hs300x_start_measurement(dev);
```

Comment:
Not sure it's worth a dedicated function?

Reply:
done